SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -386,6 +399,51 @@ TArray<Worker_InterestOverride> USpatialSender::CreateComponentInterest(AActor*
 	return ComponentInterest;
 }
 
+RPCsOnEntityCreation USpatialSender::PackQueuedRPCsForActor(TArray<TSharedRef<FPendingRPCParams>>* RPCList, AActor* Actor)
+{
+	RPCsOnEntityCreation QueuedRPCs;
+	check((RPCList != nullptr) && (Actor != nullptr));
+	if ((RPCList == nullptr) || (Actor == nullptr))
+	{
+		return QueuedRPCs;
+	}
+
+	UClass* Class = Actor->GetClass();
+	const FClassInfo& Info = ClassInfoManager->GetOrCreateClassInfoByClass(Class);
+
+	for (TSharedRef<FPendingRPCParams> RPCParams : *RPCList)
+	{
+		if (!RPCParams->TargetObject.IsValid())
+		{
+			// The target object was destroyed before we could send the RPC.
+			continue;
+		}
+
+		const FRPCInfo* RPCInfo = Info.RPCInfoMap.Find(RPCParams->Function);
+		FUnrealObjectRef TargetObjectRef(PackageMap->GetUnrealObjectRefFromNetGUID(PackageMap->GetNetGUIDFromObject(Actor)));
+		if (TargetObjectRef != FUnrealObjectRef::UNRESOLVED_OBJECT_REF)
+		{
+			TSet<TWeakObjectPtr<const UObject>> UnresolvedObjects;
+			FSpatialNetBitWriter PayloadWriter(PackageMap, UnresolvedObjects);
+
+			if (GetDefault<USpatialGDKSettings>()->bCheckRPCOrder)
+			{
+				if (RPCParams->Function->HasAnyFunctionFlags(FUNC_NetReliable) && !RPCParams->Function->HasAnyFunctionFlags(FUNC_NetMulticast))
+				{
+					PayloadWriter << RPCParams->ReliableRPCIndex;
+				}
+			}
+
+			TSharedPtr<FRepLayout> RepLayout = NetDriver->GetFunctionRepLayout(RPCParams->Function);
+			RepLayout_SendPropertiesForRPC(*RepLayout, PayloadWriter, RPCParams->Parameters.GetData());
```

Comment:
This would crash if any arguments to RPCs were pointers to UObjects that got destroyed.